common/djangoapps/third_party_auth/samlutils/utils.py

```diff
@@ -0,0 +1,38 @@
+"""
+    Utility functions for use in SAMLProviderConfig, SAMLProviderData code and tests
+"""
+
+from edx_rest_framework_extensions.auth.jwt.cookies import jwt_cookie_name
+from edx_rest_framework_extensions.auth.jwt.tests.utils import (
+    generate_jwt_token,
+    generate_unversioned_payload,
+)
+from enterprise.models import EnterpriseCustomer
+
+
+def _jwt_token_from_role_context_pairs(user, role_context_pairs):
+    """
+    Generates a new JWT token with roles assigned from pairs of (role name, context).
+    """
+    roles = []
+    for role, context in role_context_pairs:
+        role_data = '{role}'.format(role=role)
+        if context is not None:
+            role_data += ':{context}'.format(context=context)
+        roles.append(role_data)
+
+    payload = generate_unversioned_payload(user)
+    payload.update({'roles': roles})
+    return generate_jwt_token(payload)
+
+
+def set_jwt_cookie(client, user, role_context_pairs=None):
+    """
+    Set jwt token in cookies
+    """
+    jwt_token = _jwt_token_from_role_context_pairs(user, role_context_pairs or [])
+    client.cookies[jwt_cookie_name()] = jwt_token
```

Comment:
These functions are explicitly for unit testing, so it would be better to put them over in this file https://github.com/edx/edx-platform/blob/c9f376c6012ca64ec8da42a8b3fcbd4f91ba3783/common/djangoapps/third_party_auth/tests/utils.py